### DISCUSSION TITLE
code for t150 works fine on Tumbleweed with sim 'Live for Speed' (LFS); used under wine

### DISCUSSION THREAD

**Discussion by AelblerBL** [scarburato/t150_driver#18] [General]
Hello and a big 'thank you' for the coding!

Briefly about me; I'm not a programmer or similarly inclined 'nerd', but just a simple Linux user ... ;-)

I found your code on git-hub. I couldn't manage the installation of the code and the necessary preparations without help (despite your instructions).
With the help of some guys from [LUG-V.s.](https://www.lug-vs.org) I managed to get the steering wheel t150 working under Linux Tumbleweed 64bit. I am convinced that this is possible under any Linux.

I use the steering wheel in the racing simulation Live for Speed (LFS), in which steering and pedals do not need to be configured; they are automatically detected. Only the steering wheel buttons have to be assigned.
Pedals and steering wheel work perfectly and I am really happy now.

[Here](https://www.lug-vs.org/lugvswiki/index.php/Lenkrad) I created a wiki (will still translate it from German into readable English).

A little anecdote:
I bought the Thrustmaster t150 steering wheel just a few weeks ago and could only use it on Windows (which I don't like to boot at all).
I then created a ticket with the manufacturer asking for support.
The answer was literally, "You will never get the steering wheel to work on Linux."
This annoyed me bit and I searched the net and found your code. Great!

As I learned [here](https://github.com/scarburato/t150_driver#what-is-missing-), I can't rely on firmware updates, and in fact it seems that after a major OS update of my Tumbleweed, a reinstallation of the code is necessary. But this is only a minor issue.

Thank you very much for your super work! Thumbs up!

Translated with www.DeepL.com/Translator (free version)

**Comment by culebron** [comment]
Vielen Danken für dieser Meldung! This is how I found about the driver.

**Reply by wowogiengen** [reply]
Hello Dimitri,
i am a close friend to AelblerBL, and I have helped him in writing his wiki entry at [the wiki of LUG-VS](https://www.lug-vs.org/lugvswiki/index.php/Lenkrad). 
We encountered a couple of issues, that could be fixed in the install script, but i dont know how to propose thos changes here. 
So I will write the issues and add my changed version of the script (included in install.zip) here:

1. the source directories may be already present when trying to create them, so I uncommented the line with `rm -rf /usr/src/t150-*`.
2. dkms tries to remove `/lib/modules/5.11.6-1-default/kernel/drivers/input/joystick/`, but this directory is not empty.
3. git clone complains about the target directory not being empty

I changed the attached install.sh so, that it is clearer for unexperienced users of linux to see where and why it fails. Non-catastrophic errors could be suppressed, so the users does not have to worry about them.

Greetings 
Wolfgang

[install.zip](https://github.com/scarburato/t150_driver/files/6248869/install.zip)

**Comment by albfan** [comment]
I suggest to turn this discussion into (how to build t150 driver)

Most (if not all) of the complaints are already solved on master
